ex2.py

```diff
@@ -0,0 +1,26 @@
+def group_by(func, iterable):
+    """Returns a dictonary that the keys are the values that are returned from the passed function
+       and the values of each key is list of all elements that retrund from the function
+       the same value as the key
+            Args:
+                func (method): the function we will preform on each object in the iterable.
+                iterable.
+            Returns:
+                dictonary: The grouped lists.
+            Raises:
+                none
+            Examples:
+                givin a list of strings and the len function
+                will return a dictornary the the keys are the length of each string
+                and the values are list of strings the the value of it equal the key.
+                >>> print(group_by(len, ["hi", "bye"]))
+                {2: ['hi'], 3: ['bye']}
```

Comment:
Write documentation in the right format - 
python
def count_hi_v4(word: str) -> int:
    """
    The function counts how many times the string "hi" appears in a given string.
    @param word: The given string.
    @return: The amount of times the word "hi" appeared.
    """
    return word.count("hi")